framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp

```diff
@@ -162,6 +162,128 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
     return status;
 }
 
+int cfg_uci_get_wireless_from_ifname(enum paramType type, char *interface_name, const char param[],
+                                     char *value)
+{
+    bool is_section_found = false;
+    struct uci_ptr ptr;
+    struct uci_context *ctx;
+    struct uci_package *p = NULL;
+    struct uci_element *e;
+    struct uci_section *s;
+    struct uci_element *n;
+    char lookup_str[MAX_UCI_BUF_LEN] = "wireless";
+
+    ctx = uci_alloc_context();
+    if (!ctx)
+        return RETURN_ERR;
+
+    if ((uci_lookup_ptr(ctx, &ptr, lookup_str, true) != UCI_OK)) {
+        ERROR("%s, uci lookup failed!\n", __FUNCTION__);
+        uci_free_context(ctx);
+        return RETURN_ERR;
+    }
+
+    p = ptr.p;
```

Comment:
Probably should make sure that `ptr` and `ptr.p` are valid.
```C++
if (!ptr || !ptr.p) { ... }
```

Reply:
fixed